File: wcm-bpm-service/camunda/modeshape-jcr-rest-boot/src/main/java/com/bpwizard/wcm/repo/rest/jcr/model/AuthoringTemplate.java
package com.bpwizard.wcm.repo.rest.jcr.model;

import java.util.Arrays;
import java.util.Map;

import com.bpwizard.wcm.repo.rest.modeshape.model.HasName;

public class AuthoringTemplate extends ResourceNode implements HasName {
	private String name;
	private String repository;
	private String workspace;
	private String library;
	private String baseResourceType;
	
	private BaseFormGroup[] formGroups;
	private Map<String, FormControl> formControls;
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public BaseFormGroup[] getFormGroups() {
		return formGroups;
	}
	public void setFormGroups(BaseFormGroup[] formGroups) {
		this.formGroups = formGroups;
	}
	
	public String getBaseResourceType() {
		return baseResourceType;
	}
	public void setBaseResourceType(String baseResourceType) {
		this.baseResourceType = baseResourceType;
	}
	public Map<String, FormControl> getFormControls() {
		return formControls;
	}
	public void setFormControls(Map<String, FormControl> formControls) {
		this.formControls = formControls;
	}
	public String getRepository() {
		return repository;
	}
	public void setRepository(String repository) {
		this.repository = repository;
	}
	public String getWorkspace() {
		return workspace;
	}
	public void setWorkspace(String workspace) {
		this.workspace = workspace;
	}
	public String getLibrary() {
		return library;
	}
	public void setLibrary(String library) {
		this.library = library;
	}
	@Override
	public String toString() {
		return "AuthoringTemplate [name=" + name + ", repository=" + repository + ", workspace=" + workspace
				+ ", library=" + library + ", baseResourceType=" + baseResourceType + ", formGroups="
				+ Arrays.toString(formGroups) + ", formControls=" + formControls + ", toString()=" + super.toString()
				+ "]";
	}
}

File: ui/angular/projects/bpw-wcm-service/src/lib/model/AuthoringTemplate.ts
import { HasName } from "./HasName";
import { BaseFormGroup } from "./BaseFormGroup";
import { FormControl } from "./FormControl";
import { ResourceNode } from "./ResourceNode";
import { AccessControlList } from "./AccessControlList";
import { AuthoringTemplateProperties } from "./AuthoringTemplateProperties";
import { WcmAuthority } from "./WcmAuthority";
export interface AuthoringTemplate extends ResourceNode, HasName {
  repository: string;
  workspace: string;
  library: string;
  baseType?: string;
  nodeType?: string;
  contentItemAcl?: AccessControlList;
  elementGroups?: BaseFormGroup[];
  elements?: { [key: string]: FormControl };
  properties?: AuthoringTemplateProperties;
  checkedOut?: boolean;
  wcmAuthority?: WcmAuthority;
}
